app/models/nic/base.rb

```diff
@@ -358,6 +360,12 @@ def interface_attribute_uniqueness(attr, base = Nic::Base.where(nil))
       db_candidates = db_candidates.select { |c| c.id != id && in_memory_candidates.map(&:id).include?(c.id) }
       errors.add(attr, :taken) if db_candidates.present?
     end
+
+    def check_blank_mac_for_virtual_resources
+      if host.compute_provides?(:mac) && mac
```

Comment:
I'd prefer to be more verbose here:
```suggestion
      if host.compute_provides?(:mac) && !mac.empty?
```

Reply:
```suggestion
      if host.compute_provides?(:mac) && mac.present?
```